finalfrontier/src/vocab.rs

```diff
@@ -213,10 +213,29 @@ impl From<SubwordVocab> for VocabWrap {
         .into()
     }
 }
+/// Index representation.
+///
+/// This enum represents possible types of indices, e.g. in a vocabulary with subwords, the `Multi`
+/// variant includes all indices associated with a key.
+#[derive(Clone)]
+pub enum Idx {
+    Single(usize),
```

Comment:
Should this also be `u64` or should both just be `usize`, what's the motivation for `u64` in the indexing methods?

Reply:
The `u64` comes from hashing in subword indices, since we use a 64-bit hash. I have to think a bit about this. In Rust-land, there are no obvious benefits, since on a 32-bit machine we have to index on arrays at some point using usize and it is unlikely that we'll have more than 2^32 embedding array rows anytime soon. However, `u64` is nice for a C ABIs because `usize` does not have an equivalent integer type in C outside `uintptr_t`, whereas `u64` is equivalent to `uint64_t`.

Reply:
Both enum variants should have wrap the same type. So, I guess both `usize` for now.